indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -259,15 +302,16 @@ private synchronized void startScheduling()
 
     private synchronized void stopScheduling()
     {
-      status = ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_SHUTDOWN;
+      state = ScheduledBatchSupervisor.State.SUSPENDED;
+      statusTracker.cleanupStaleTaskStatuses();
```

Comment:
Since this is being done inside the task manager of a single supervisor, this method should accept the supervisor ID and cleanup items only for that supervisor.

Alternatively, don't call this method from inside the `ScheduledBatchTask` and instead schedule a periodic cleanup job on the `tasksExecutor`.

Reply:
Yes, good point, updated the method `cleanupStaleTaskStatuses()` to accept a supervisor ID and it now cleans up only tasks associated with the supplied supervisor ID.